src/DSi.cpp

```diff
@@ -884,19 +961,32 @@ u32 ARM9Read32(u32 addr)
         break;
 
     case 0x03000000:
+    case 0x03800000:
         if (addr >= NWRAMStart[0][0] && addr < NWRAMEnd[0][0])
         {
             u8* ptr = NWRAMMap_A[0][(addr >> 16) & NWRAMMask[0][0]];
+#if DSI_NWRAM_DEBUGOUT
+            printf("Set A9: Read %08x from NWRAM at %08x (source=%08x)\n", ptr ? *(u32*)&ptr[addr & 0xFFFF] : 0, addr, ptr);
+            fflush(stdout);
```

Comment:
this shouldn't be necessary

Reply:
Given that the access massively changed, This allows observing the accesses to verify it along the testcase.
Since this is guarded at compile time, It is not included in the build unless forced to.